src/hotspot/share/opto/superword.cpp

```diff
@@ -2811,11 +2811,8 @@ bool SuperWord::is_vector_use(Node* use, int u_idx) const {
   }
 
   if (VectorNode::is_muladds2i(use)) {
-    // MulAddS2I takes shorts and produces ints.
-    if (u_pk->size() * 2 != d_pk->size()) {
-      return false;
-    }
-    return true;
+    assert(1 <= u_idx && u_idx <= 4, "expected range for the 4 inputs");
+    return _packset.is_muladds2i_pack_with_pack_inputs(u_pk);
```

Comment:
Do we really need this assert? `is_muladds2i()` checks that use is MulAddS2I which by definition has 4 data inputs + control.

Reply:
I can remove it.